docs/source/advanced/pruning_quantization.rst

```diff
@@ -0,0 +1,106 @@
+.. testsetup:: *
+
+    import os
+    from pytorch_lightning.trainer.trainer import Trainer
+    from pytorch_lightning.core.lightning import LightningModule
+
+.. _pruning_quantization:
+
+########################
+Pruning and Quantization
+########################
+
+Pruning and Quantization are techniques to compress model size for deployment, allowing inference speed up and energy saving without significant accuracy losses.
+
+*******
+Pruning
+*******
+
+.. warning::
+
+     Pruning is in beta and subject to change.
+
+Pruning is focussing on eliminating some of the model weights to reduce the model size and increase inference performance. Pruning has been shown to achieve significant efficiency improvements while minimizing the drop in model performance (prediction quality).
+
+Pruning your model is recommended for cloud endpoint, on edge device, or mobile deployment.
+
+To enable pruning during training in Lightning, simply pass in the :class:`~pytorch_lightning.callbacks.ModelPruning` callback to the Lighting Trainer (using native PyTorch pruning implementation under the hood).
+
+This callback suports multiple pruning functions: pass any `torch.nn.utils.prune <https://pytorch.org/docs/stable/nn.html#utilities>`_ function as a string to select which weights to pruned (`random_unstructured <https://pytorch.org/docs/stable/generated/torch.nn.utils.prune.random_unstructured.html#torch.nn.utils.prune.random_unstructured>`_, `RandomStructured <https://pytorch.org/docs/stable/generated/torch.nn.utils.prune.RandomStructured.html#torch.nn.utils.prune.RandomStructured>`_, etc) or implement your own by subclassing `BasePruningMethod <https://pytorch.org/tutorials/intermediate/pruning_tutorial.html#extending-torch-nn-utils-prune-with-custom-pruning-functions>`_.
+
+TODO: what do you have to set?
```

Comment:
The user would have to provide the `amount` to be pruned. The amount can be a float between 0 and 1 or a function which takes the current_epoch and return a float. In the later case, the sum of all returned amount should be smaller than 1. One can return 0 or None to skip pruning for a given epoch.

Reply:
Should we really specify what to set here? It's all in the pruning class docs.

Reply:
No, you are right.